File: haskell/fizzbuzz.hs
main :: IO ()
main = printAll $ map fizzbuzz [1..100]

printAll :: [String] -> IO()
printAll = foldr ((>>) . putStrLn) (return ())

fizzbuzz :: Int -> String
fizzbuzz x | x `mod` 15 == 0 = "fizzbuzz"
           | x `mod` 3  == 0 = "fizz"
           | x `mod` 5  == 0 = "buzz"
           | otherwise       = show x

-- vim: expandtab shiftwidth=4 softtabstop=4

File: ruby/lib/fizzbuzz.rb
#!/usr/bin/env ruby

class FizzBuzz
    def input(number)
        if number.modulo(15) == 0
            "fizzbuzz"
        elsif number.modulo(3) == 0
            "fizz"
        elsif number == 5
            "buzz"
        else
            number
        end
    end
end

for i in 1..100
    if i.modulo(3) == 0
        puts "Fizz"
    elsif i.modulo(5) == 0
        puts "Buzz"
    elsif i.modulo(15) == 0
        puts "FizzBuzz"
    else
        puts i
    end
end

# vim: expandtab shiftwidth=4 softtabstop=4
